x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_creation/components/configuration_step/analysis_fields_table.tsx

```diff
@@ -67,141 +69,141 @@ const columns = [
 
-export const MemoizedAnalysisFieldsTable: FC<{
-  excludes: string[];
+export const AnalysisFieldsTable: FC<{
+  includes: string[];
   loadingItems: boolean;
-  setFormState: any;
+  setFormState: React.Dispatch<React.SetStateAction<any>>;
   tableItems: FieldSelectionItem[];
-}> = memo(
-  ({ excludes, loadingItems, setFormState, tableItems }) => {
-    const [sortableProperties, setSortableProperties] = useState();
-    const [currentSelection, setCurrentSelection] = useState<any[]>([]);
+}> = ({ includes, loadingItems, setFormState, tableItems }) => {
+  const [sortableProperties, setSortableProperties] = useState();
+  const [currentPageIndex, setCurrentPageIndex] = useState<number>(0);
 
-    useEffect(() => {
-      if (excludes.length > 0) {
-        setCurrentSelection(excludes);
-      }
-    }, [tableItems]);
+  useEffect(() => {
+    if (includes.length === 0 && tableItems.length > 0) {
+      const includedFields: string[] = [];
+      tableItems.forEach((field) => {
+        if (field.is_included === true) {
+          includedFields.push(field.name);
+        }
+      });
+      setFormState({ includes: includedFields });
+    } else if (includes.length > 0) {
+      setFormState({ includes });
+    }
+  }, [tableItems]);
 
-    // Only set form state on unmount to prevent re-renders due to props changing if exludes was updated on each selection
-    useEffect(() => {
-      return () => {
-        setFormState({ excludes: currentSelection });
-      };
-    }, [currentSelection]);
+  useEffect(() => {
+    let sortablePropertyItems = [];
+    const defaultSortProperty = 'name';
 
-    useEffect(() => {
-      let sortablePropertyItems = [];
-      const defaultSortProperty = 'name';
+    sortablePropertyItems = [
+      {
+        name: 'name',
+        getValue: (item: any) => item.name.toLowerCase(),
+        isAscending: true,
+      },
+      {
+        name: 'is_included',
+        getValue: (item: any) => item.is_included,
+        isAscending: true,
+      },
+      {
+        name: 'is_required',
+        getValue: (item: any) => item.is_required,
+        isAscending: true,
+      },
+    ];
+    const sortableProps = new SortableProperties(sortablePropertyItems, defaultSortProperty);
 
-      sortablePropertyItems = [
-        {
-          name: 'name',
-          getValue: (item: any) => item.name.toLowerCase(),
-          isAscending: true,
-        },
+    setSortableProperties(sortableProps);
+  }, []);
+
+  const filters = [
+    {
+      type: 'field_value_selection',
+      field: 'is_included',
+      name: i18n.translate('xpack.ml.dataframe.analytics.create.excludedFilterLabel', {
+        defaultMessage: 'Is included',
+      }),
+      multiSelect: false,
+      options: [
         {
-          name: 'is_included',
-          getValue: (item: any) => item.is_included,
-          isAscending: true,
+          value: true,
+          view: (
+            <EuiText grow={false}>
```

Comment:
The Yes and No options seem to swap around depending on what I have selected. I would expect 'Yes' to always be shown first looking at the `options` list. Is this because `isAscending` was removed?

![image](https://user-images.githubusercontent.com/7405507/86119975-642cb380-bacb-11ea-8c60-92f44f41ae21.png)

![image](https://user-images.githubusercontent.com/7405507/86119989-6b53c180-bacb-11ea-928c-067199be7e67.png)

Reply:
Switched to EUI field_value_toggle_group in 6dfc33aa26986266df2924e5f6d13286b38da3be which should remove this problem